### PR TITLE
NMS-14267: Add smoke test for FileEditor

### PR DESCRIPTION
**Pull request by chuklov**
FileEditor automation test that tests regular and wrong behaviour of the user. 

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14267

### CODE REVIEW

File: smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
+package org.opennms.smoketest.rest;
```
Comment (synqotik) [inline smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java:1]: Need license header, can copy from another file. May sure to update copyright date to `1999-2022`

File: smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+    }
+
+    /**
+     * Test that the File Editor will deny file that is not in the correct format
+     */
+    @Test
+    public void wrongFileName() {
```
Comment (synqotik) [inline smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java:141]: May want a more specific method name (and log message and comment), for example `unsupportedFileExtension`, comment: "Test that the File Editor will deny a file with an unsupported file extension".

File: smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+    }
+
+    /**
+     * Test that the File Editor will deny changes that are not in the correct format
+     */
+    @Test
+    public void wrongUpdate() {
```
Comment (synqotik) [inline smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java:164]: May want a more specific method name / comment, for example `updateFailsOnXmlValidation`, "Test that the File Editor will deny changes when invalid XML is uploaded for an XML file"

**Review by synqotik — CHANGES_REQUESTED** [review]
Looks good! Just a few comments to address.

**Review by synqotik — DISMISSED** [review]
LGTM!

File: smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        }
+
+        // remove the file
+        MediaType mediaType = MediaType.parse("text/plain");
+        body = RequestBody.create(mediaType, "");
+        try {
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + FILE_NAME, "DELETE", body);
+            Assert.assertEquals(200, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
```
Comment (cgorantla) [inline smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java:161]: Add more context to all the  LOG

File: smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+    }
+
+    /**
+     * API request to create a user
+     * @param request
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    protected Integer doRequest(final HttpRequestBase request) throws IOException, InterruptedException {
```
Comment (cgorantla) [inline smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java:284]: Now that you got that working without `AbstractOpenNMSSeleniumHelper`. Can you try to reuse those methods from `AbstractOpenNMSSeleniumHelper` by making them public static or by Abstracting them in another class

File: smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+    @Before
+    public void setUp() throws InterruptedException {
+
+        RestAssured.baseURI = STACK.opennms().getBaseUrlExternal().toString();
+        RestAssured.port = STACK.opennms().getWebPort();
+        RestAssured.authentication = preemptive().basic(BASIC_AUTH_USERNAME, BASIC_AUTH_PASSWORD);
+
+        addUserAPI();
+
+        Thread.sleep(5000);
```
[the remark this answers is not in the document]
Comment (chuklov) [reply smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java:108]: Unfortunately yes, as if there is no sleep, the test failed as the user is not yet available.

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**